src/client/containers/learnerGallery/index.jsx

```diff
@@ -6,65 +6,56 @@ import _ from 'lodash';
 class LearnerGallery extends Component {
   constructor(props) {
     super(props);
-
-    if (this.props.type) {
-      this.state = {
-        selectedLearners: this.filterByType(this.props.type),
-      };
-    } else {
-      let searchSkill = this.props.match.params.searchSkill.replace(/search=/, '').split(',');
-      this.state = {
-        selectedLearners: this.filterByMultipleSkills(searchSkill),
-      };
-    }
-
+    this.state = {
+      searchBar: '',
+    };
     this.handleChange = this.handleChange.bind(this);
   }
 
   handleChange(event) {
     event.preventDefault();
-    this.setState({ selectedLearners: event.target.value });
+    this.setState({ searchBar: event.target.value });
   }
 
   filterByName () {
-    let filteredLearner;
-    if (this.props.type) {
-      filteredLearner = this.filterByType(this.props.type);
-    } else {
-      let searchSkill = this.props.match.params.searchSkill.replace(/search=/, '').split(',');
-      filteredLearner = this.filterByMultipleSkills(searchSkill);
-    }
-    if (Array.isArray(this.state.selectedLearners)) {
+    const filteredLearner = this.determineSubsetOfLearners(this.props.type);
+
+    if (!this.state.searchBar) {
       return filteredLearner;
     }
-    let searchTerm = this.state.selectedLearners.toLowerCase().split();
-    let learnersBySkill = this.filterByOneSkill(searchTerm);
+
+    const searchTerm = this.state.searchBar.toLowerCase().split(' ')[0];
     const foundLearners = filteredLearner.filter(learner => {
-      return learner.name.toLowerCase().includes(this.state.selectedLearners.toLowerCase());
+      return learner.name.toLowerCase().includes(searchTerm.toLowerCase());
```

Comment:
this change reduces complexity by pushing the search term into a variable that's local to the function, (rather than the local state of the component), right?